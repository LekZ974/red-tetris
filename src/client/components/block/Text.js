import styled from 'react-emotion'
import {
  fontSize,
  space,
  color,
  responsiveStyle,
  lineHeight,
  textAlign
} from 'styled-system'
import theme from '../../theme'

export const caps = props =>
  props.caps
    ? {
      textTransform: 'uppercase'
    }
    : null

export const regular = props =>
  props.regular ? { fontWeight: props.theme.regular } : null

export const bold = props =>
  props.bold ? { fontWeight: props.theme.bold } : null

export const italic = props => (props.italic ? { fontStyle: 'italic' } : null)

const align = responsiveStyle('text-align', 'align')

const Text = styled('div')`
  ${italic} ${fontSize} ${space} ${color} ${caps} ${regular} ${bold} ${align} ${textAlign} ${lineHeight};
`

Text.displayName = 'Text'

Text.defaultProps = {
  theme: theme,
  color: 'text'
}

Text.span = Text.withComponent('span')
Text.p = Text.withComponent('p')
Text.s = Text.withComponent('s')

export default Text
