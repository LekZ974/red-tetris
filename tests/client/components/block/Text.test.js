import React from 'react';
import Text from '../../../../src/client/components/block/Text';
import renderer from 'react-test-renderer';

test('Text display', () => {
  const component = renderer.create(
    <Text>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Text bold display', () => {
  const component = renderer.create(
    <Text bold>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Text regular display', () => {
  const component = renderer.create(
    <Text regular>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Text uppercase display', () => {
  const component = renderer.create(
    <Text caps>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Text italic display', () => {
  const component = renderer.create(
    <Text italic>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Text with wrong props display', () => {
  const component = renderer.create(
    <Text wrongprops>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Text with multiple props display', () => {
  const component = renderer.create(
    <Text italic bold caps regular>Hello</Text>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
