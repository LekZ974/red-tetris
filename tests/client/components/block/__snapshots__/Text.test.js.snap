// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text bold display 1`] = `
<div
  className="css-aibj26"
  color="text"
>
  Hello
</div>
`;

exports[`Text display 1`] = `
<div
  className="css-d726wp"
  color="text"
>
  Hello
</div>
`;

exports[`Text italic display 1`] = `
<div
  className="css-1hfam3m"
  color="text"
>
  Hello
</div>
`;

exports[`Text regular display 1`] = `
<div
  className="css-11leol5"
  color="text"
>
  Hello
</div>
`;

exports[`Text uppercase display 1`] = `
<div
  className="css-165tkgl"
  color="text"
>
  Hello
</div>
`;

exports[`Text with multiple props display 1`] = `
<div
  className="css-1mkoec2"
  color="text"
>
  Hello
</div>
`;

exports[`Text with wrong props display 1`] = `
<div
  className="css-d726wp"
  color="text"
>
  Hello
</div>
`;
